d2-schemas/src/main/pegasus/com/linkedin/d2/D2Service.pdl

```diff
@@ -63,15 +69,24 @@ record D2Service includes D2ChangeTimeStamps {
   cluster: string
 
   /**
-   * The load balancer strategy properties for this service
+   * The load balancer strategy properties for this service.
+   *
+   * Required along with D2DegraderProperties for the degrader strategy.
    */
   loadBalancerStrategyProperties: D2LoadBalancerStrategyProperties
 
   /**
-   * The degrader properties for this service
+   * The degrader properties for this service.
+   *
+   * Required along with D2DegraderProperties for the degrader strategy.
    */
   degraderProperties: D2DegraderProperties
```

Comment:
Do we need to mark these 2 old fields as optional since when client migrates, they can only set relativeStrategyProperties and get rid of these 2 old property fields

Reply:
Wondering why this not optional even today, if user specify RANDOM, then these properties are not used either. Is it not optional for a purpose?

Reply:
I assume it's because any normal use case will have these fields. Random is really for testing I think. I will mark these 2 as optional.